runtime/router.go

```diff
@@ -250,74 +150,28 @@ func (router *HTTPRouter) Register(
 	)
 }
 
-func (router *HTTPRouter) handleNotFound(w http.ResponseWriter, r *http.Request) {
-	resFields := []zapcore.Field{
-		zap.Int(statusCodeZapName, 404),
-	}
-	writeLogs(router.gateway.Logger, logRequestFields(r), resFields)
+func (router *HTTPRouter) handleNotFound(
+	w http.ResponseWriter,
+	r *http.Request,
+) {
+	req := NewServerHTTPRequest(w, r, nil, router.notFoundEndpoint)
 	// TODO custom NotFound
-	// A NotFound request is not started...
-	// TODO: inc.finish()
 	http.NotFound(w, r)
+	req.res.writeHeader(http.StatusNotFound)
```

Comment:
why do we have to call `writeHeader` ? and can we do so before `http.NotFound()` ?

Reply:
Actually, I just need to set the `req.res.StatusCode`, so that it gets picked up by `finish()`.  Removing call to `writeHeader()`